sdk/core/core-amqp/src/ConnectionContextBase.ts

```diff
@@ -177,7 +177,7 @@ export const ConnectionContextBase = {
     }
 
     const connectionOptions: ConnectionOptions = {
-      transport: Constants.TLS,
+      transport: (parameters.config.useDevelopmentEmulator ? Constants.TCP : Constants.TLS) as any,
```

Comment:
is the `as any` cast needed? Looking at the rhea connection options definition it seems to allow both "tls" and "tcp". with "tcp" optional when it's Net Connection

https://github.com/amqp/rhea/blob/8fa5eb545e52732ee95e4a88597e19b7ec62bf05/typings/connection.d.ts#L219

Reply:
You're right. This is just a case of TS not working hard enough to analyze the type. I think the intersection type is what made TS stop checking, see this small repro: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgBoEZkG8BQzkAeAXMgETqk4C+OOoksiKqATNnoSaS5TTmAE8ADs2QBeNJgA+aFgG5agkcgCa4tMgBk2ZAJIBnMFFABzZHwA2EMMgBG6+Bf0QFOBAHsQhzmom58xHbIAPxkFMhcPAA01EA

Reply:
I'm also confused about what TS is doing here, a repro closer to the original: https://www.typescriptlang.org/play?#code/FDCWDsBcFMCcDMCGBjaACAKsgDgZQPbIDW0kAwvuONMpAPLaQDOaA3sGmtvrJAFxpwAVwC2AIzgBuYAF8wUOElRoAktmQFipClRr1GLdp2yJIACwFNIsCAHNpc4JACe2dADltlarQbM0ALyYOJok5N56fiwAPqrqoV66vgbS8jAIKB6kGLCI4EzcvH6glIYcaNZ5BTz8aABEkDh1DiAQ6UroGAA2TDlVhfqQJfls5ZX5AwINPc3lk4KiErAtIC5uaJ7kXaDQUDo+Q5TFpYEb2bkTNccjAGRn4UmDTKlr6Pt6w9csQZtk27sPA6fRjDGKYHp9S5FEGlVLAZClSBoBGPYGHfICd60NGg05GCoXaq8AQACmsQnQAH56o1sHU0FNIDMAJQAGjmNQEAEYAEwAZgALOyZJIgA

Reply:
Ah right, my sample shows a weird error message but expected behavior.